### ISSUE TITLE
Structured logging support

### ISSUE THREAD

**Issue by slkinnison** [giantswarm/kubernetes-elastic-stack#17]
Add support for structured logging via Elastic stack. This would require adding json_decode_fields support (see https://github.com/elastic/beats/commit/50db534c873e165aa5cc76e77c4fc25e1a320872). Ideally/selfishly I would like to see a recipe for structured logging of a Spring Boot application leveraging https://github.com/logstash/logstash-logback-encoder#encoder.

**Comment by webwurst** [comment]
@slkinnison We are already using [`json` decode](https://www.elastic.co/guide/en/beats/filebeat/master/configuration-filebeat-options.html#config-json) for the log events that are sent out by the Docker daemon. If your application itself uses `json` for logging you would want to setup an [`ingest`](https://www.elastic.co/guide/en/beats/filebeat/master/configuring-ingest-node.html) and pipe events to that, i guess.

**Comment by webwurst** [comment]
Ok, with [release](https://www.elastic.co/blog/beats-5-1-1-released) of version `v5.1.1` there is a:
> new processor, called decode_json_fields, that can decode JSON from an arbitrary field.

So you could edit the `ConfigMap` for `Filebeat` accordingly. But still, using `ingest` instead makes probably also sense.

**Comment by slkinnison** [comment]
FYI: I did try updating the filebeat image to v.5.1.1 and use the decode_json_fields, but I got `WARN Can not index event (status=400): {"type":"mapper_parsing_exception","reason":"failed to parse [json.log]","caused_by":{"type":"illegal_state_exception","reason":"Can't get text on a START_OBJECT at 1:154"}}`. Using "log" as the value of decode_json_fields had no effect as expected since that is not the name of the field.

I will look into configuring an ingest pipeline.

Thanks again for the pointers.
